ad-joining/register-computer/main.py

```diff
@@ -289,6 +323,41 @@ def __register_computer(request):
         logging.exception("Checking project access to '%s' failed" % auth_info.get_project_id())
         return flask.abort(HTTP_ACCESS_DENIED)
     
+    # If custom OU is being used, make sure all provided OUs (root and computer) are valid    
+    if "CUSTOM_OU_ROOT_DN" in os.environ:
+        try:
+            computer_ou = None
+            # The service is configured to use custom OUs. Make sure the root OU is valid
+            custom_ou_root = __read_required_setting("CUSTOM_OU_ROOT_DN")
+            logging.debug("Service is configured to use custom OU '%s'" % custom_ou_root)
+            if __is_custom_ou_valid(ad_connection, custom_ou_root):
+                # Locate the customer OU for the computer and make sure it is valid
+                custom_target_ou = __get_computer_ou(gce_instance)
+                if custom_target_ou and __is_custom_ou_valid(ad_connection, custom_target_ou):
+                    logging.debug("Found custom OU '%s' for compute instance '%s' in project '%s'" 
+                        % (custom_target_ou, auth_info.get_instance_name(), auth_info.get_project_id()))
+
+                    # Verify the OU provided for the computer is a descendant of the custom root OU
+                    if custom_target_ou.endswith(custom_ou_root):
```

Comment:
Should that be `startswith`?

Reply:
It should be endswith, as the root is higher up the tree, which means the target includes the parent, prefixed by the OU path from root to target. For example:
Root - OU=AllBUs,OU=Cloud,DC=gmsa,DC=lab
Target - OU=BU1,OU=AllBUs,OU=Cloud,DC=gmsa,DC=lab

(target string ends with root string)

I will however add .lower() to both strings, in case of nonidentical casing